x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/cohere/embeddings/CohereEmbeddingsServiceSettings.java

```diff
@@ -107,7 +91,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.startObject();
 
         commonSettings.toXContentFragment(builder);
-        builder.field(EMBEDDING_TYPE, embeddingType);
+        builder.field(EMBEDDING_TYPE, elementType());
```

Comment:
The parser is reading with `CohereEmbeddingType::fromString` but `DenseVectorFieldMapper.ElementType` is written here. It works because they are both byte/float but it seems pretty fragile

Reply:
Yep, thanks for catching that.